src/Runner.Worker/JobRunner.cs

```diff
@@ -110,6 +110,8 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
                 Directory.CreateDirectory(toolsDirectory);
                 jobContext.SetRunnerContext("tool_cache", toolsDirectory);
 
+                jobContext.SetRunnerContext("version", BuildConstants.RunnerPackage.Version);
```

Comment:
can we remove this for now? i think this is worth wider discussion first. There are a few different approaches to solve the problem. Maybe a small ADR to discuss the merits of each why we chose one over the others.

Reply:
+1